app/routes/planet_routes.py

```diff
@@ -0,0 +1,94 @@
+from flask import Blueprint, abort, make_response, request, Response
+from app.routes.db import db
+from app.models.planet import Planet
+
+
+planets_bp = Blueprint("planets_bp", __name__, url_prefix="/planets")
+
+
+@planets_bp.post("")
+def create_planet():
+
+    request_body = request.get_json()
+    name = request_body["name"]
+    description = request_body["description"]
+    distance_from_sun = request_body["distance_from_sun"]
+
+    new_planet = Planet(
+        name=name, description=description, distance_from_sun=distance_from_sun
+    )
+    db.session.add(new_planet)
+    db.session.commit()
+
+    response = new_planet.to_dict()
+
+    return response, 201
+
+@planets_bp.get("")
+def get_all_planets():
+    query = db.select(Planet)
+    name_param = request.args.get("name")
+
+    if name_param:
+        query = query.where(Planet.name == name_param)
+
+    description_param = request.args.get("description")
+    if description_param:
+        query = query.where(Planet.description.ilike(f"%{description_param}%"))
+
+    distance_from_sun_param = request.args.get("distance_from_sun")
+    if distance_from_sun_param:
+        query = query.where(Planet.distance_from_sun.ilike(f"%{distance_from_sun_param}%"))
+
+    query = query.order_by(Planet.id)
+
+    planets = db.session.scalars(query)
+
+    planets_response = [planet.to_dict() for planet in planets]
+    return planets_response
```

Comment:
Like `create_model` we could also move this logic into a helper function like `get_models_with_filters`. How could the helper function look different to serve your needs here?